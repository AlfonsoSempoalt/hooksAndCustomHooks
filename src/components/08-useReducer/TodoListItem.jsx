import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ todo, handleToggle, handleOnDelete, index }) => {
  return (
    <>
      <li key={todo.id} className='list-group-item text-center'>
        <p
          className={`${todo.done && 'complete'}`}
          onClick={() => handleToggle(todo.id)}
        >  
          {index + 1} {todo.desc}
        </p>
        <button
          className='btn btn-danger'
          onClick={() => handleOnDelete(todo.id)}
        >
          Delete
        </button>
      </li>
    </>
  );
};

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleOnDelete: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
};

export default TodoListItem;
