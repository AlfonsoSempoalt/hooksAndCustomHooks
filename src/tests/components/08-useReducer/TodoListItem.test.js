import React from 'react';
import { shallow } from 'enzyme';
import { demoTodos } from '../fixtures/demoTodos';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';

describe('tests to <TodoListItem />', () => {
    const handleToggle = jest.fn();
    const handleOnDelete = jest.fn();

    let wrapper = shallow(
        <TodoListItem
            todo={demoTodos[0]}
            handleToggle={handleToggle}
            handleOnDelete={handleOnDelete}
            index={0}
        />
    );

    beforeEach(() => {
        wrapper = shallow(
            <TodoListItem
                todo={demoTodos[0]}
                handleToggle={handleToggle}
                handleOnDelete={handleOnDelete}
                index={0}
            />
        );
    });

    test('should match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call the function handleDelete', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(handleOnDelete).toHaveBeenCalled();
    });

    test('should call the function handleToggle', () => {
        wrapper.find('p').at(0).simulate('click');
        const IDTODO = 1;
        expect(handleToggle).toHaveBeenCalledWith(IDTODO);
    });

    test('should show the content of the todo', () => {
        const todo = demoTodos[0];
        todo.done = true;
        wrapper = shallow(
            <TodoListItem
                todo={todo}
                handleToggle={handleToggle}
                handleOnDelete={handleOnDelete}
                index={0}
            />
        );

        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    });
});
