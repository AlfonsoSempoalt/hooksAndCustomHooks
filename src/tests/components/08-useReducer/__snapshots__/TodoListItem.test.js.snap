// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <TodoListItem /> should match with the snapshot 1`] = `
<Fragment>
  <li
    className="list-group-item text-center"
    key="1"
  >
    <p
      className="false"
      onClick={[Function]}
    >
      1
       
      aprender react
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>
</Fragment>
`;
